ffmpeg-plugin/mcm_rx.c

```diff
@@ -0,0 +1,220 @@
+#include "libavutil/log.h"
+#include "libavutil/opt.h"
+#include "libavutil/internal.h"
+#include "libavutil/pixdesc.h"
+#include "libavformat/avformat.h"
+#include "libavformat/mux.h"
+#include "libavformat/internal.h"
+#include <mcm_dp.h>
+#include <bsd/string.h>
+
+typedef struct McmDemuxerContext {
+    const AVClass *class; /**< Class for private options. */
+
+    /* arguments */
+    char *ip_addr;
+    char *port;
+    char *payload_type;
+    char *protocol_type;
+    int width;
+    int height;
+    enum AVPixelFormat pixel_format;
+    AVRational frame_rate;
+    char *socket_name;
+    int interface_id;
+
+    mcm_conn_context *rx_handle;
+    bool first_frame;
+} McmDemuxerContext;
```

Comment:
I oppose this as it is creating another structure with the same parameters that are already defined in sub-directory `sdk/include`
, for example structure `mcm_conn_context`. Why not extend it with missing parameters or even moving this one there.

Reply:
Agree. But let's do the refactoring in a separate PR. I suggest not to amend this one.